source/AI.cpp

```diff
@@ -520,6 +520,12 @@ void AI::Step(const PlayerInfo &player, Command &activeCommands)
 			value = min(FENCE_MAX, value + FENCE_DECAY + 1);
 		}
 	
+	// How many roving ships there are in the system.
+	unsigned int rovingShipCount = 0;
+	for(const auto &ship: ships)
```

Comment:
Iterating through every ship *again* is less than ideal. But you're in luck: the for statement a few lines above *also* iterates through every ship. Fold them please. :)